redhat-developer/web-terminal-tooling
# Update tooling for OCP 4.16; update base images

This PR updates the tooling used in the web terminal operator tooling container to their latest versions available on mirror.openshift.com and github releases for OCP 4.16, as well as updates the base container images used to the latest UBI9-minimal images.

We had to update to UBI9 based images from UBI8 since  oc 4.16 requires GLIBC_2.33 & GLIBC_2.34 which are only available on RHEL 9 based images.

The curl-minimal package is also provided by default in the UBI9 image, and installing
curl manually results in package conflicts. Thus we are no longer manually installing
curl in the tooling container.

# Testing
For convenience, I've pushed the web terminal tooling container built from this PR to `quay.io/aobuchow/wto-tooling:1.11`.
1. Launch an OpenShift 4.16+ cluster. I recommend OpenShift 4.17, since the [backport to fix using kubeadmin with WTO](https://github.com/redhat-developer/web-terminal-operator/issues/162#issuecomment-2206400246) has not been merged to OpenShift 4.16 yet. 
2. Install the web-terminal operator from Operator Hub
3. Create a web terminal, click the link to specify a custom image, and provide `quay.io/aobuchow/wto-tooling:1.11`. Alternatively, create a web terminal with the default image and run `wtoctl set image quay.io/aobuchow/wto-tooling:1.11` in the web terminal console
4. Verify the web terminal starts up successfully. The provided binaries should be updated to the versions specified in this PR. For example, running `oc version` should indicate version 4.16.

Here's the output of `help` from the tooling image:

```
the OpenShift Web Terminal. Type "help" for a list of installed CLI tools.
bash-5.1 ~ $ help
Installed tools:
Command         Version                          Name
oc              4.16.0                           OpenShift CLI
kubectl         1.29.1                           Kubernetes CLI
kustomize       5.4.2                            Kustomize CLI
helm            3.14.4                           Helm CLI
kn              1.12.0                           KNative CLI
tkn             0.37.0                           Tekton CLI
subctl          0.17.2                           Submariner CLI
odo             3.16.1                           
v3.16.1         Red Hat OpenShift Developer CLI  
virtctl         1.2.2                            KubeVirt CLI
rhoas           0.53.0                           Red Hat OpenShift Application Services CLI
jq              1.6                              jq
```
